src/pudl/output/ferc1.py

```diff
@@ -3044,25 +3084,112 @@ def out_ferc1__yearly_rate_base(
-    # then select only the leafy exploded records that are in rate base and concat
-    in_rate_base = pd.concat(
+    return cash_working_capital
+
+
+def breakdown_unlabeled(
+    rate_base: pd.DataFrame,
+    split_col: str,
+) -> pd.DataFrame:
+    """Breakdown rate base records with an unlabeled split_col.
+
+    Args:
+        rate_base: full table of rate base data.
+        split_col: column with the label that contains unlabeled values
+            (ex: null or total).
+    """
+    # the works for both the util type and the in rate base tags because
+    # for both tags: unlabeled is represented as a null and but just util type:
+    # unlabeled is also represented as "total"
+    unlabeled_mask = (rate_base[split_col] == "total") | rate_base[split_col].isnull()
+    ratio_idx = ["report_year", "utility_id_ferc1"]
+    ratio_df = get_split_col_ratio(
+        # remove the unlabeled records because total is the value
+        # we want to breakdown so we can't have it in the columns to sum up
+        rate_base[~unlabeled_mask],
+        ratio_idx=ratio_idx,
+        split_col=split_col,
+    )
+    rate_base_broken_down = apply_ratio_to_breakdown_unlabeled(
+        rate_base=rate_base,
+        ratio_df=ratio_df,
+        ratio_idx=ratio_idx,
+        unlabeled_mask=unlabeled_mask,
+        split_col=split_col,
+    )
+    return rate_base_broken_down.convert_dtypes()
+
+
+def get_split_col_ratio(rate_base_labeled, ratio_idx, split_col) -> pd.DataFrame:
+    """Calculate the percentage of the total labeled ``ending_balance`` within each tag group.
+
+    Make ratio column with a 0-1 value of the sum of ``ending_balance`` in each labled
+    ``split_col`` within ``ratio_idx``.
+    """
+    # get the sum of the balance in each of the values in split_col
+    rate_base_grouped = rate_base_labeled.groupby(ratio_idx + [split_col])[
+        ["ending_balance"]
+    ].sum(min_count=1)
+    df = rate_base_grouped.reset_index(level=[split_col]).pivot(columns=[split_col])
+    split_values = df.columns.get_level_values(1)
+    df.columns = split_values
+    df["abs_summed"] = abs(df).sum(axis=1)
+    for split_value in split_values:
+        df[f"ratio_{split_value}"] = abs(df[split_value]) / abs(df.abs_summed)
+    ratio = pd.DataFrame(
+        df.filter(regex="^ratio_").stack(future_stack=False),
+        columns=[f"ratio_{split_col}"],
+    ).reset_index()
+    ratio[split_col] = ratio[split_col].str.removeprefix("ratio_")
+    assert all(
+        ~ratio[f"ratio_{split_col}"].between(0, 1, inclusive="both")
+        | ratio[f"ratio_{split_col}"].notnull()
+    )
+    return ratio
+
+
+def apply_ratio_to_breakdown_unlabeled(
+    rate_base: pd.DataFrame,
+    ratio_df: pd.DataFrame,
+    unlabeled_mask: pd.Series,
+    ratio_idx: list[str],
+    split_col: str,
+):
+    """For the unlabeled tag records, broadcast labeled tags and allocate the ``ending_balance``.
+
+    Apply ratios from :func:`get_split_col_ratio` to ``ending_balance``.
```

Comment:
Does this mean that you're allocating `ending_balance` based on the ratio assigned to that row?

Reply:
yes! I ended up removing this whole function and moving this content up into the main function above and explaining it above.